jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -627,76 +481,143 @@ inline bool JERRY_ATTR_ALWAYS_INLINE
-} /* ecma_builtin_regexp_prototype_symbol_replace */
-
-/**
- * The RegExp.prototype object's '@@search' routine
- *
- * See also:
- *          ECMA-262 v6.0, 21.2.5.9
+ * Dispatcher of the Regexp built-in's routines
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
-static ecma_value_t
-ecma_builtin_regexp_prototype_symbol_search (ecma_value_t this_arg, /**< this argument */
-                                             ecma_value_t string_arg) /**< string argument */
+ecma_value_t
+ecma_builtin_regexp_prototype_dispatch_routine (uint16_t builtin_routine_id, /**< built-in wide routine
+                                                                              *   identifier */
+                                                ecma_value_t this_arg, /**< 'this' argument value */
+                                                const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                                        *   passed to routine */
+                                                ecma_length_t arguments_number) /**< length of arguments' list */
 {
-  return ecma_regexp_search_helper (this_arg, string_arg);
-} /* ecma_builtin_regexp_prototype_symbol_search */
+  JERRY_UNUSED (arguments_number);
 
-/**
- * The RegExp.prototype object's '@@split' routine
- *
- * See also:
- *          ECMA-262 v6.0, 21.2.5.11
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_regexp_prototype_symbol_split (ecma_value_t this_arg, /**< this argument */
-                                            ecma_value_t string_arg, /**< source string */
-                                            ecma_value_t limit_arg) /**< limit */
-{
-  return ecma_regexp_split_helper (this_arg, string_arg, limit_arg);
-} /* ecma_builtin_regexp_prototype_symbol_split */
+  /* 1. */
+  if (!ecma_is_value_object (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("'this' is not an object"));
+  }
 
-/**
- * The RegExp.prototype object's '@@match' routine
- *
- * See also:
- *          ECMA-262 v6.0, 21.2.5.6
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_regexp_prototype_symbol_match (ecma_value_t this_arg, /**< this argument */
-                                            ecma_value_t string_arg) /**< source string */
-{
-  return ecma_regexp_match_helper (this_arg, string_arg);
-} /* ecma_builtin_regexp_prototype_symbol_match */
+#if !ENABLED (JERRY_ESNEXT)
+  bool require_regexp = builtin_routine_id <= ECMA_REGEXP_PROTOTYPE_ROUTINE_TO_STRING;
+#else /* ENABLED (JERRY_ESNEXT) */
+  bool require_regexp = builtin_routine_id < ECMA_REGEXP_PROTOTYPE_ROUTINE_TO_STRING;
 #endif /* ENABLED (JERRY_ESNEXT) */
 
+  if (require_regexp && !ecma_object_is_regexp_object (this_arg))
```

Comment:
`this_arg` is already an object (checked above) `ecma_object_class_is (obj_p, LIT_MAGIC_STRING_REGEXP_UL)` is enough.